http/http2/src/main/java/io/helidon/http/http2/Http2ConnectionWriter.java

```diff
@@ -164,27 +170,26 @@ public int writeHeaders(Http2Headers headers,
      * @throws InterruptedException in case we fail to lock on the stream
      */
     public void updateHeaderTableSize(long newSize) throws InterruptedException {
-        withStreamLock(() -> {
+        lock();
+        try {
             outboundDynamicTable.protocolMaxTableSize(newSize);
-            return null;
-        });
+        } finally {
+            streamLock.unlock();
+        }
     }
 
     private void lockedWrite(Http2FrameData frame) {
-        withStreamLock(() -> {
+        lock();
+        try {
             noLockWrite(frame);
-            return null;
-        });
+        } finally {
+            streamLock.unlock();
+        }
     }
 
-    private <T> T withStreamLock(Callable<T> callable) {
+    private void lock() {
```

Comment:
Why is this refactored?

Reply:
Hi, please excuse the comment from the peanut gallery, but I recently stumbled over the rationale why such a method wasn't added to the `Lock` interface itself: https://bugs.openjdk.org/browse/JDK-8025597?focusedCommentId=13414177&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-13414177